### PR TITLE
Update master to solana v1.17

### PR DESCRIPTION
**Pull request by buffalu**
- Pull in v1.17 code to master
- Move to all dependencies in Cargo.toml
- Remove jito-solana dependency
- Update some geyser code

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -5,3 +5,49 @@ members = [
...
+authors = ["Jito Foundation <team@jito.network>"]
+edition = "2021"
+repository = "https://github.com/jito-foundation/geyser-grpc-plugin"
+homepage = "https://jito.network/"
+
+[workspace.dependencies]
+bincode = "1.3.3"
+bs58 = "0.5.0"
+clap = { version = "4.4.6", features = ["derive", "env"] }
+clap_lex = "0.5.1"
```
Comment (esemeniuc) [inline Cargo.toml:24]: prob can remove this since clap is upgraded

File: Cargo.toml [hunk 2]
```diff
@@ -5,3 +5,49 @@ members = [
...
+solana-logger = "1.17.1"
+solana-metrics = "1.17.1"
+solana-program = "1.17.1"
+solana-sdk = "1.17.1"
+solana-transaction-status = "1.17.1"
+solana-vote-program = "1.17.1"
+thiserror = "1.0.40"
+tokio = { version = "1.29.1", features = ["rt-multi-thread"] }
+tokio-stream = "0.1"
+toml_datetime = "=0.6.0" # manual dep override for clap.
```
Comment (esemeniuc) [inline Cargo.toml:50]: ditto

File: Cargo.toml [hunk 3]
```diff
@@ -5,3 +5,49 @@ members = [
...
+tonic = { version = "0.10.2", features = ["tls"] }
+tonic-build = "0.10.2"
```
Comment (esemeniuc) [inline Cargo.toml:52]: dev dep?

File: proto/proto/confirmed_block.proto [hunk 4]
```diff
@@ -72,7 +72,18 @@ message TransactionError {
 
 message InnerInstructions {
     uint32 index = 1;
-    repeated CompiledInstruction instructions = 2;
+    repeated InnerInstruction instructions = 2;
+}
+
+message InnerInstruction {
```
Comment (esemeniuc) [inline proto/proto/confirmed_block.proto:78]: would be nice to have a blurb about how this would be used

File: proto/src/convert.rs [hunk 5]
```diff
@@ -770,8 +804,10 @@ impl TryFrom<tx_by_addr::TransactionError> for TransactionError {
             27 => TransactionError::InvalidRentPayingAccount,
             28 => TransactionError::WouldExceedMaxVoteCostLimit,
             29 => TransactionError::WouldExceedAccountDataTotalLimit,
-            // 32 => TransactionError::BundleNotContinuous,
-            // 33 => TransactionError::SkippedExecution,
+            32 => TransactionError::MaxLoadedAccountsDataSizeExceeded,
+            33 => TransactionError::InvalidLoadedAccountsDataSizeLimit,
+            34 => TransactionError::ResanitizationNeeded,
+            36 => TransactionError::UnbalancedTransaction,
```
Comment (esemeniuc) [inline proto/src/convert.rs:810]: 35 missing?

Comment (esemeniuc) [reply proto/src/convert.rs:810]: nvm on line 768

File: Cargo.toml (hunk 3 above)
Comment (buffalu) [reply Cargo.toml:52]: no top level workspace dev dependencies

File: proto/proto/confirmed_block.proto (hunk 4 above)
Comment (buffalu) [reply proto/proto/confirmed_block.proto:78]: this folder is 99% copied from v1.17 solana labs repository

**Review by esemeniuc — APPROVED** [review]

**Pull request merged**